src/compiler/visitor.ts

```diff
@@ -549,6 +549,25 @@ namespace ts {
         return <Statement>singleOrUndefined(nodes) || createBlock(<NodeArray<Statement>>nodes);
     }
 
+    export function visitIterationBody(body: Statement, visitor: Visitor, context: TransformationContext): Statement {
+        context.startBlockScope();
+        const updated = visitNode(body, visitor, isStatement, liftToBlock);
+        const declarations = context.endBlockScope();
+        if (some(declarations)) {
+            if (isBlock(updated)) {
+                declarations.push(...updated.statements);
+            }
+            else {
+                declarations.push(updated);
+            }
+            return setTextRange(

```

Comment:
Take a look at `visitFunctionBody` in _visitorPublic.ts_. If `updated` is a block (as on line 558) you should use `updateBlock` (so that any `EmitFlags` are carried over) and only use `createBlock` when creating a new `Block` (as would be the case on line 561).